openssl/aes.go

```diff
@@ -296,6 +296,12 @@ func (c *aesCipher) NewGCM(nonceSize, tagSize int) (cipher.AEAD, error) {
 	return c.newGCM(false)
 }
 
+// NewGCMTLS returns a GCM cipher which is specific to TLS
+// and should not be used outside of that context.
```

Comment:
This is nitpicky, but `which` kind of implies that all GCM ciphers are specific to TLS. `that` or dropping the word entirely makes it clearer IMO.

(Also suggesting a change to make the context more explicit while I'm here, but I think that line would be fine either way:)
```suggestion
// NewGCMTLS returns a GCM cipher specific to TLS
// and should not be used for non-TLS purposes.
```